### PR TITLE
plat-stm32mp1: assign last 4kB of sysram as shared memory

### PR DESCRIPTION
**Pull request by etienne-lms**
Allow the last 4kByte of stm32mp1 SYSRAM internal RAM to be
assigned to non-secure world when used as SCMI shared memory.
ETZPC memory firewall is configured accordingly from service
late initialization level as ETPCZ driver is initialized from
service init level when embedded BTD support is enabled.

Platform configuration switches CFG_STM32MP1_SCMI_SHM_BASE and
CFG_STM32MP1_SCMI_SHM_SIZE are used to define the SCMI shared
memory location.

Compilation asserts that if CFG_TZSRAM_START is define inside SYSRAM
then it fully resides inside the secure SYSRAM area as per SoC ETZPC
implementation that mandates the non-secure SYSRAM to be above (higher
address) the secure SYSRAM.

Signed-off-by: Etienne Carriere <etienne.carriere@st.com>

### CODE REVIEW

File: core/arch/arm/plat-stm32mp1/main.c [hunk 1]
```diff
@@ -211,24 +211,22 @@ void main_secondary_init_gic(void)
 	stm32mp_register_online_cpu();
 }
 
-#ifndef CFG_EMBED_DTB
 static TEE_Result init_stm32mp1_drivers(void)
 {
+#ifndef CFG_EMBED_DTB
 	/* Without secure DTB support, some drivers must be inited */
 	stm32_etzpc_init(ETZPC_BASE);
+#endif
```
Comment (jforissier) [inline core/arch/arm/plat-stm32mp1/main.c:219]: Minor: `if (!IS_ENABLED(CFG_EMBED_DTB))`?

Comment (etienne-lms) [reply core/arch/arm/plat-stm32mp1/main.c:219]: right, I should get used to this way. Will fix (and wll try to remember for the next times).

**Comment by jforissier** [comment]
`Acked-by: Jerome Forissier <jerome@forissier.org>`

**Comment by etienne-lms** [comment]
tag applied

**Pull request merged**